homeassistant/components/hive/light.py

```diff
@@ -39,6 +39,21 @@ async def async_setup_entry(
 class HiveDeviceLight(HiveEntity, LightEntity):
     """Hive Active Light Device."""
 
+    def __init__(self, hive, hive_device):
+        """Initialise hive light."""
+        super().__init__(hive, hive_device)
+        if self.device["hiveType"] == "warmwhitelight":
+            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+        elif self.device["hiveType"] == "tuneablelight":
+            self._attr_supported_color_modes = {ColorMode.COLOR_TEMP}
+        elif self.device["hiveType"] == "colourtuneablelight":
+            self._attr_supported_color_modes = {ColorMode.COLOR_TEMP, ColorMode.HS}
+            rgb = self.device["status"].get("hs_color")
+            self._attr_hs_color = color_util.color_RGB_to_hs(*rgb)
```

Comment:
We don't really need to set the hs color since that will be done by the update method during entity addition.

Reply:
This is a copy and paste error when moving from the update function to the __init__ one.